src/dmd/root/filename.d

```diff
@@ -631,71 +631,74 @@ nothrow:
+            version (Windows)
             {
-                const(char)* p = FileName.path(path);
-                if (*p)
+                size_t len = strlen(path);
+                if ((len > 2 && p[-1] == ':' && strcmp(path + 2, p) == 0) || len == strlen(p))
                 {
-                    version (Windows)
-                    {
-                        size_t len = strlen(path);
-                        if ((len > 2 && p[-1] == ':' && strcmp(path + 2, p) == 0) || len == strlen(p))
-                        {
-                            mem.xfree(cast(void*)p);
-                            return 0;
-                        }
-                    }
-                    bool r = ensurePathExists(p);
                     mem.xfree(cast(void*)p);
-
-                    if (r)
-                        return r;
-                }
-                version (Windows)
-                {
-                    char sep = '\\';
-                }
-                else version (Posix)
-                {
-                    char sep = '/';
-                }
-                if (path[strlen(path) - 1] != sep)
-                {
-                    version (Windows)
-                    {
-                        int r = _mkdir(path);
-                    }
-                    version (Posix)
-                    {
-                        int r = mkdir(path, (7 << 6) | (7 << 3) | 7);
-                    }
-                    if (r)
-                    {
-                        /* Don't error out if another instance of dmd just created
-                         * this directory
-                         */
-                        version (Windows)
-                        {
-                            // see core.sys.windows.winerror - the reason it's not imported here is because
-                            // the autotester's dmd is too old and doesn't have that module
-                            enum ERROR_ALREADY_EXISTS = 183;
-
-                            if (GetLastError() != ERROR_ALREADY_EXISTS)
-                                return true;
-                        }
-                        version (Posix)
-                        {
-                            if (errno != EEXIST)
-                                return true;
-                        }
-                    }
+                    return true;
                 }
             }
+            bool r = ensurePathExists(p);
+            mem.xfree(cast(void*)p);
+
+            if (!r)
+                return r;
+        }
+
+        version (Windows)
+            int r = _mkdir(path);
+        version (Posix)
+            int r = mkdir(path, (7 << 6) | (7 << 3) | 7);
+
+        if (r == 0)
+            return true;
+
+        // Don't error out if another instance of dmd just created
+        // this directory
+        version (Windows)
+        {
+            // see core.sys.windows.winerror - the reason it's not imported here is because
```

Comment:
Now that we have github, core.sys.windows.winerror can be productively replaced with https://github.com/dlang/druntime/blob/master/src/core/sys/windows/winerror.d#L160